src/utils/bootstrapUtils.js

```diff
@@ -130,8 +129,45 @@ export function getClassSet(props) {
   return classes;
 }
 
-export function omitBsProps(props) {
-  return omit(props, ['bsClass', 'bsSize', 'bsStyle']);
+function getBsProps(props) {
+  return {
+    bsClass: props.bsClass,
+    bsSize: props.bsSize,
+    bsStyle: props.bsStyle,
+  };
+}
+
+function isBsProp(propName) {
+  return (
+    propName === 'bsClass' ||
+    propName === 'bsSize' ||
+    propName === 'bsStyle'
+  );
+}
+
+export function splitBsProps(props) {
+  const elementProps = {};
+  Object.entries(props).forEach(([propName, propValue]) => {
+    if (!isBsProp(propName)) {
+      elementProps[propName] = propValue;
+    }
+  });
+
+  return [getBsProps(props), elementProps];
+}
+
+export function splitBsPropsAndOmit(props, omittedPropNames) {
```

Comment:
This should be a slight optimization... just two statically separate functions rather than a branch in the same function. Figure it's worth it since we call this literally everywhere.